src/Compilers/VisualBasic/Portable/BoundTree/Statement.vb

```diff
@@ -0,0 +1,992 @@
+                Return Me.Condition
+            End Get
+        End Property
+
+        Protected Overrides Function StatementKind() As OperationKind
+            Return OperationKind.IfStatement
+        End Function
+
+    End Class
+
+    Partial Class BoundSelectStatement
+        Implements ISwitchStatement
+
+        Private ReadOnly Property ICases As ImmutableArray(Of ICase) Implements ISwitchStatement.Cases
+            Get
+                Return Me.CaseBlocks.As(Of ICase)()
+            End Get
+        End Property
+
+        Private ReadOnly Property IValue As IExpression Implements ISwitchStatement.Value
+            Get
+                Return Me.ExpressionStatement.Expression
+            End Get
+        End Property
+
+        Protected Overrides Function StatementKind() As OperationKind
+            Return OperationKind.SwitchStatement
+        End Function
+    End Class
+
+    Partial Class BoundCaseBlock
+        Implements ICase
+
+        Private ReadOnly Property IBody As ImmutableArray(Of IStatement) Implements ICase.Body
+            Get
+                Return ImmutableArray.Create(Of IStatement)(Me.Body)
+            End Get
+        End Property
+
+        Private ReadOnly Property IClauses As ImmutableArray(Of ICaseClause) Implements ICase.Clauses
+            Get
+                If Me.CaseStatement.CaseClauses.IsEmpty Then
+                    Return ImmutableArray.Create(CaseElseClause)
+                End If
+
+                Return Me.CaseStatement.CaseClauses.As(Of ICaseClause)()
+            End Get
+        End Property
+
+        Protected Overrides Function StatementKind() As OperationKind
+            Return OperationKind.None
+        End Function
+
+        Private Shared CaseElseClause As ICaseClause = New CaseElse
+
+        Private Class CaseElse
+            Implements ICaseClause
+
+            Private ReadOnly Property ICaseClass As CaseKind Implements ICaseClause.CaseKind
+                Get
+                    Return CaseKind.Default
+                End Get
+            End Property
+        End Class
+
+    End Class
+
+    Partial Class BoundCaseClause
+        Implements ICaseClause
+
+        Protected MustOverride ReadOnly Property ICaseClass As CaseKind Implements ICaseClause.CaseKind
+    End Class
+
+    Partial Class BoundSimpleCaseClause
+        Implements ISingleValueCaseClause
+
+        Private ReadOnly Property IEquality As BinaryOperationKind Implements ISingleValueCaseClause.Equality
+            Get
+                Dim caseValue As BoundExpression = DirectCast(Me.IValue, BoundExpression)
+                If caseValue IsNot Nothing Then
+                    Select Case caseValue.Type.SpecialType
+                        Case SpecialType.System_Int32, SpecialType.System_Int64, SpecialType.System_UInt32, SpecialType.System_UInt64, SpecialType.System_UInt16, SpecialType.System_Int16, SpecialType.System_SByte, SpecialType.System_Byte, SpecialType.System_Char
```

Comment:
I don't think so, but that is a fine question for future consideration. I'll add a comment.